daemon/redis.c

```diff
@@ -812,7 +726,6 @@ static int json_get_hash(struct redis_hash *out, struct call* c,
 	out->ht = g_hash_table_new_full(g_str_hash, g_str_equal, free, freeReplyObject);
```

Comment:
The hash table values aren't redis reply objects any more, so using `freeReplyObject` as value free function is incorrect.

Several points here.

The `redis_hash->rr` pointer used to contain the redis object containing the entire hash (in array form) for string storage. Replacing an entire hash/array with a single string makes no sense. So the `redis_hash` struct doesn't need to contain a string, but rather should contain only a `GHashTable`. I'll make separate comments on all relevant code lines.

For now I think the best approach is to `str_dup` the string values into the hash table (as you do it now) and use `free` as the value free function. (Since `json_reader_get_string_value_uri_enc` also mallocs its buffer, this can then be improved into a single malloc - I'll do this later on)